deployer/deployer-services/src/main/java/io/hyscale/deployer/services/processor/PodParentProcessor.java

```diff
@@ -0,0 +1,120 @@
+/**
+ * Copyright 2019 Pramati Prism, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.hyscale.deployer.services.processor;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.hyscale.commons.exception.HyscaleException;
+import io.hyscale.deployer.services.factory.PodParentFactory;
+import io.hyscale.deployer.services.handler.PodParentHandler;
+import io.hyscale.deployer.services.model.PodParent;
+import io.kubernetes.client.openapi.ApiClient;
+
+public class PodParentProcessor {
```

Comment:
He is always providing the parent of the pod ?
PodParentProvider ? What do you say ?

Reply:
Yeah, that seems better.